### DISCUSSION TITLE
[30 Giugno 2022] Pipeline

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/architettura-degli-elaboratori#83] [Pipeline (M-Z)]
![Esame](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/4ebbeb2f-b6f4-4601-89fc-506385b5f9b5)
![Esame](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/597ba6cc-add4-433e-8308-fb3ca64a7257)
![Architettura Pipeline](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/8175251e-bff4-4972-be24-c9171ad5196f)

**Comment by daromacarlo** [comment]
![1](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/80e7faf5-8dba-4627-80a3-eaacace08cbf)
![2](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/a1872ba2-02c7-4c6b-b5db-d1ee0698426c)
![4](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/e71660ff-afa4-4cb7-80ed-e09e93f47253)
![5](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/27ad0333-3704-479d-ab49-3ac0718bcad0)
![67](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/cd88b147-cbfd-4998-8194-c56b72a63cae)

sono ben accette correzioni in casso di errore :)

EDIT:  Mi sono dimenticato di inserire nel conto dei cicli senza farwarding 2 stalli dovuti ad un data hazard nella fase di pre-ciclo, il totale dei cicli è quindi 67, non 65.

**Comment by Alegau03** [comment]
> ![1](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/80e7faf5-8dba-4627-80a3-eaacace08cbf)
> 
> ![2](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/a1872ba2-02c7-4c6b-b5db-d1ee0698426c)
> 
> ![4](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/e71660ff-afa4-4cb7-80ed-e09e93f47253)
> 
> ![5](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/27ad0333-3704-479d-ab49-3ac0718bcad0)
> 
> ![67](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/cd88b147-cbfd-4998-8194-c56b72a63cae)
> 
> 
> 
> sono ben accette correzioni in casso di errore :)
> 
> 
> 
> EDIT:  Mi sono dimenticato di inserire nel conto dei cicli senza farwarding 2 stalli dovuti ad un data hazard nella fase di pre-ciclo, il totale dei cicli è quindi 67, non 65.
> 
> 

Hai scordato 2 stalli nella fase senza forwarding tra 17 e 18

**Comment by daromacarlo** [comment]
@Alegau03 controlla se ti esce così:
C.hazard & D.hazard
![Cattgggggggggggggggggura](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/0dcd1fdb-837f-4504-b08f-c804d1f2a29e)

![Catturqqqqqqqqqqqqqqa](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/7c037b0e-b516-49f7-ae43-94483a595983)
cicli senza forwarding = 79

**Reply by thegeek-sys** [reply]
il ciclo uscente non sono 2 istruzioni e 3 data hazard nella parte senza forwarding?
<img width="1249" alt="immagine" src="https://github.com/sapienzastudentsnetwork/architettura-degli-elaboratori/assets/55345633/54fccc72-41f5-4a53-bd9f-a0c0e66479cd">

**Comment by Alegau03** [comment]
> @Alegau03 controlla se ti esce così:
> 
> C.hazard & D.hazard
> 
> ![Cattgggggggggggggggggura](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/0dcd1fdb-837f-4504-b08f-c804d1f2a29e)
> 
> 
> 
> ![Catturqqqqqqqqqqqqqqa](https://github.com/sapienzastudentsnetwork/archelab2223/assets/135441131/7c037b0e-b516-49f7-ae43-94483a595983)
> 
> cicli senza forwarding = 79
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

Si mi tornano, mentre con forwarding sono 61

**Comment by daromacarlo** [comment]
@Alegau03 eh sì, ho contato il ciclo (+) una volta

**Comment by RaulComaz** [comment]
Sul data Hazard il "8-9-$t0" non è da contare in quanto nella riga 8 $t0 non viene modificato

**Comment by daromacarlo** [comment]
@GrenzGaenger450 dipende quale dei due programmi stai analizzando, il programma "nero" ha righe 8/9 invertite rispetto ha programma "bianco"

**Comment by RaulComaz** [comment]
non avevo notato questo errore di scrittura del prof
Grazie per avermelo fatto notare

**Comment by daromacarlo** [comment]
@GrenzGaenger450 prego
